ImageList/APIClient/APIService.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  UniversalServise.swift
+//  ImageList
+//
+//  Created by Александр Зиновьев on 13.02.2023.
+//
+
+import Foundation
+
+protocol APIServiceProtocol {
+    func fetch<T: Decodable>(
+        request: UnsplashRequests,
+        expectedType type: T.Type,
+        completion: @escaping (Result<T, Error>) -> Void
+    )
+}
+
+final class APIService: APIServiceProtocol {
+    private var task: URLSessionTask?
+    
+    func fetch<T: Decodable>(
+        request: UnsplashRequests,
+        expectedType type: T.Type,
+        completion: @escaping (Result<T, Error>) -> Void
+    ) {
+        assert(Thread.isMainThread, "Thread.isMainThread")
+        task?.cancel()
+
+        let task = URLSession.shared.objectTask(
+            for: request.request,
+            expected: type) { result in
+                switch result {
+                case .success(let body):
+                    completion(.success(body))
+                    self.task = nil
+                case .failure(let error):
+                    completion(.failure(error))
+                    self.task = nil
+                }
```

Comment:
в остальном коде это дублирование учтено, а вот здесь видимо пропустил:
```suggestion
                switch result {
                case .success(let body):
                    completion(.success(body))
                case .failure(let error):
                    completion(.failure(error))
                }
                self.task = nil
```

Reply:
не понимаю) что нужно сделать) Добавить self.task = nil после switch? а в Нутри удалить?

Reply:
Здесь если видно на suggest'е  self.task = nil делается один раз, а у тебя два раза. При этом я видела похожие конструкции в других местах у тебя, и такой ошибки там нет :)

Reply:
спасибо понял)